### ISSUE TITLE
stream=True through python openai lib crashes the proxy

### ISSUE THREAD

**Issue by ankrgyl** [BerriAI/litellm#103]
```python
import openai
openai.api_base = 'http://localhost:5000'

for resp in openai.ChatCompletion.create(model="gpt-3.5-turbo", messages=[{"role": "user", "content": "Hello world"}], stream=True):
    print(resp)
```

The server prints:

```
Traceback (most recent call last):
  File "/Users/ankur/projects/liteLLM-proxy/venv/lib/python3.10/site-packages/waitress/channel.py", line 428, in service
    task.service()
  File "/Users/ankur/projects/liteLLM-proxy/venv/lib/python3.10/site-packages/waitress/task.py", line 168, in service
    self.execute()
  File "/Users/ankur/projects/liteLLM-proxy/venv/lib/python3.10/site-packages/waitress/task.py", line 472, in execute
    self.write(chunk)
  File "/Users/ankur/projects/liteLLM-proxy/venv/lib/python3.10/site-packages/waitress/task.py", line 316, in write
    towrite += data + b"\r\n"
TypeError: unsupported operand type(s) for +: 'OpenAIObject' and 'bytes'
```

**Comment by ghost** [comment]
Hi @ankrgyl - thanks for the code. Just so I have the context, Does your proxy support streaming? And are you returning OpenAIObjects? 

On the surface - the stacktrace error doesn't look like one triggered by litellm.

Does this work for you when you use the openai-python sdk on a colab?

**Comment by ishaan-jaff** [comment]
Taking a look - going to try to repro this issue on our end

**Comment by ankrgyl** [comment]
@krrishdholakia this is using your proxy (https://github.com/BerriAI/litellm/blob/main/cookbook/proxy-server/readme.md)

**Comment by ishaan-jaff** [comment]
@ankrgyl found the root cause of the error: 
TLDR: looks like it's caused because we're not handling the generator object returned by OpenAI correctly:

Currently adding this to the proxy, fixes the error for my on local
```
def generate_responses(response):
    for chunk in response:
        yield json.dumps({"response": chunk}) + "\n"
```

Pushing a fix rn

**Comment by ishaan-jaff** [comment]
Fix is pushed here: https://github.com/BerriAI/liteLLM-proxy 

Here's how i made the request with streaming:
```
import requests
import json

url = "http://localhost:5000/chat/completions"

payload = json.dumps({
  "model": "gpt-3.5-turbo",
  "stream": "True",
  "messages": [
    {
      "content": "Hello, whats the weather in San Francisco??",
      "role": "user"
    }
  ]
})
headers = {
  'Content-Type': 'application/json'
}

response = requests.request("POST", url, headers=headers, data=payload)

print(response.text)

```

Here's the streamed response:

<img width="391" alt="Screenshot 2023-08-12 at 10 34 45 AM" src="https://github.com/BerriAI/litellm/assets/29436595/31e65efb-49cd-44bb-9db8-7456140429b9">

@ankrgyl lmk if it works for you now 

also as a heads up - I was using `litellm.completion` not `opeai.chatCompletion`

**Comment by ankrgyl** [comment]
I'm using the openai lib (a requirement for this project). It no longer crashes, but it also does not seem to stream. Here's a comparison (first one uses litellm proxy, second one does not):

![Screen Recording 2023-08-12 at 11 07 39 AM](https://github.com/BerriAI/litellm/assets/565363/1bdeacf3-91ec-4f9f-8be3-b2bb041c214d)

**Comment by ishaan-jaff** [comment]
did I understand this correctly 

you're setting `openai.api_base = "https://localhost:5001"` -> which i'm assuming is the liteLLM proxy server and when you make a `chatCompletion()` request to the proxy server with `stream=True` it does not stream the response back ? 

- does it work fine without streaming ?

**Comment by ghost** [comment]
would it be helpful to do this over a call @ankrgyl / @ishaan-jaff ?

**Comment by ishaan-jaff** [comment]
I was able to repro this issue locally @ankrgyl thanks for the gif 

<img width="1065" alt="Screenshot 2023-08-12 at 11 26 39 AM" src="https://github.com/BerriAI/litellm/assets/29436595/b15b5fc5-3248-4da5-911d-907d169043ab">

will fix and revert back

**Comment by ishaan-jaff** [comment]
@ankrgyl just pushed a fix on: https://github.com/BerriAI/liteLLM-proxy

Added a testfile: `test_proxy_stream.py` which repros your scenario (afaik) 

![stream_proxy](https://github.com/BerriAI/litellm/assets/29436595/ec00def3-2fb2-41e9-8c96-6424ceb8f3df)

Let me know if streaming works for you now 😊

**Comment by ghost** [comment]
closing based on email conversation. Feel free to reopen if this doesn't work for you @ankrgyl
